cmd/gossamer/config.go

```diff
@@ -465,6 +434,33 @@ func setDotGlobalConfigFromToml(tomlCfg *ctoml.Config, cfg *dot.GlobalConfig) {
 	}
 }
 
+func setDotBasepathConfig(ctx *cli.Context, tomlConfig *ctoml.Config, cfg *dot.GlobalConfig) {
+	setDotBasepathConfigFromToml(tomlConfig, cfg)
+	setDotBasepathConfigFromFlags(ctx, cfg)
+}
+
+func setDotBasepathConfigFromToml(tomlCfg *ctoml.Config, cfg *dot.GlobalConfig) {
+	if tomlCfg == nil {
+		return
+	}
+
+	if tomlCfg.Global.BasePath != "" {
+		cfg.BasePath = tomlCfg.Global.BasePath
+	}
+}
+
+func setDotBasepathConfigFromFlags(ctx *cli.Context, cfg *dot.GlobalConfig) {
+	// check --basepath flag and update node configuration
+	if basepath := ctx.GlobalString(BasePathFlag.Name); basepath != "" {
+		cfg.BasePath = basepath
+	}
+
+	// check if cfg.BasePath his been set, if not set to default
+	if cfg.BasePath == "" {
+		cfg.BasePath = dot.GssmrConfig().Global.BasePath
```

Comment:
Is this the same default base path for all `dot.*` configs?

Reply:
I don't know the answer to this question.

Reply:
No, the base path is different for `gssmr` (`~/.gossamer/gssmr`), `kusama` (`~/.gossamer/kusama`) and `polkadot` (`~/.gossamer/polkadot`).